app/src/main/java/com/fastlink/zekrapp/viewModel/ViewModel.kt

```diff
@@ -0,0 +1,79 @@
+package com.fastlink.zekrapp.viewModel
+
+import androidx.compose.runtime.State
+import androidx.compose.runtime.mutableIntStateOf
+import androidx.compose.runtime.mutableStateOf
+import androidx.lifecycle.ViewModel
+import com.fastlink.zekrapp.appData.ZekirCategorySingleton
+import com.fastlink.zekrapp.appData.ZekirSingleton
+import com.fastlink.zekrapp.appData.model.ZekirCategoryModel
+import com.fastlink.zekrapp.appData.model.ZekirModel
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+
+class MainViewModel : ViewModel() {
+    private var _zekirCategories = mutableStateOf(ZekirCategorySingleton.zekirCategoriesData)
+    val zekirNumber = mutableIntStateOf(0)
+    val zekirCounter = mutableIntStateOf(0)
+    var zekirs :  List<ZekirModel> = emptyList()
```

Comment:
Don't expose mutable variables to other classes, you don't want to make other classes to be able to modify your variables.

Reply:
i changed all this things but I didn't know why didn't come

Reply:
deleted both variables _zekirCategory and zekirCategory  instead of this i created a method getAllCategories and get data from singleton